pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -0,0 +1,504 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// NodeAgent interfaces with baseosmgr for baseos upgrade and test validation
+// we will transition through zboot for baseos upgrade validation process
+
+// nodeagent publishes the following topic
+//   * zboot config                 <nodeagent>  / <zboot> / <config>
+//   * nodeagent status             <nodeagent>  / <status>
+
+// nodeagent subscribes to the following topics
+//   * global config
+//   * base os status               <baseosmgr> / <baseos> / <status>
+//   * zboot status                 <baseosmgr> / <zboot> / <status>
+//   * ledblink config              <zedagent>  /<ledblink> / <config>
+//   * zedagent status              <zedagent>/ <status>
+
+package nodeagent
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/cast"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zboot"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	agentName             = "nodeagent"
+	maxConfigGetFailCount = 5
+)
+
+// Version : module version
+var Version = "No version specified"
+var globalConfig = types.GlobalConfigDefaults
+var rebootDelay = 30 // take a 30 second delay
+
+// XXX globals filled in by subscription handlers and read by handlemetrics
+// XXX could alternatively access sub object when adding them.
+var clientMetrics interface{}
+var logmanagerMetrics interface{}
+var downloaderMetrics interface{}
+var networkMetrics types.NetworkMetrics
+
+type nodeagentContext struct {
```

Comment:
Isn't most of this dead stuff copied from zedagent? Need to prune to what is actually used in nodeagent.

Reply:
sure, will remove these.